examples/function-listener/print-trace.go

```diff
@@ -0,0 +1,139 @@
+package print_trace
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"log"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/internal/wasm"
+)
+
+type stackKey struct{}
+
+type callListener struct {
+	message string
+}
+
+func (l *callListener) Before(ctx context.Context) context.Context {
+	currStack, _ := ctx.Value(stackKey{}).([]string)
+	return context.WithValue(ctx, stackKey{}, append(currStack, l.message))
+}
+
+func (l *callListener) After(_ context.Context) {
+}
+
+type callListenerFactory struct {
+}
+
+func (f *callListenerFactory) NewListener(info wasm.FunctionInfo) wasm.FunctionListener {
+	return &callListener{
+		message: fmt.Sprintf("%v(%v) %v", info.Name, info.Params, info.Returns),
+	}
+}
+
+// main shows how to define, import and call a Go-defined function from a
+// WebAssembly-defined function.
+//
+// See README.md for a full description.
+func main() {
+	// Choose the context to use for function calls.
+	ctx := context.Background()
+
+	// Create a new WebAssembly Runtime.
+	r := wazero.NewRuntimeWithConfig(wazero.NewRuntimeConfigInterpreter())
+
+	// Sets the FunctionListenerFactory using an internal accessor.
```

Comment:
While this example was useful for me to see the listening, realized it's not useful to anyone else since it can only be used in this repo. Guess I should remove the example before getting ready to merge

Reply:
you could move it to internal/examples for now